sunpy/net/dataretriever/client.py

```diff
@@ -225,7 +170,57 @@ def _can_handle_query(cls, *query):
         `sunpy.net.fido_factory.UnifiedDownloaderFactory`
         class uses to dispatch queries to this Client.
         """
-        raise NotImplementedError
+        adict = cls.register_values()
+        if cls.optional == set():
+            cls.optional = {k for k in adict.keys()} - cls.required
+        if not cls.check_attr_types_in_query(query, cls.required, cls.optional):
+            return False
+        for key in adict:
+            all_vals = [i[0].lower() for i in adict[key]]
+            for x in query:
+                if isinstance(x, key) and str(x.value).lower() not in all_vals:
+                    return False
+        return True
+
+    def post_search_hook(self, exdict, matchdict):
+        """
+        Helper function used after `:func:~sunpy.net.dataretriever.GenericClient.search`
+        which makes the extracted metadata representable in a query response table.
+
+        Parameters
+        ----------
+        exdict: `dict`
+            Represents metadata extracted from files.
+        matchdict: `dict`
+            Contains attr values accessed from `register_values()`
+            and the search query itself.
+
+        Returns
+        -------
+        map_: `~collections.OrderedDict`
+            An Ordered Dictionary which is used by `QueryResponse`
+            to show results.
+        """
+        map_ = OrderedDict()
+        almost_day = TimeDelta(1 * u.day - 1 * u.millisecond)
+        if 'month' in exdict and 'day' in exdict:
```

Comment:
Now I think about it, the fact there is `"year", "month" and "day"` in this dict at all is weird.

Reply:
https://github.com/sunpy/sunpy/pull/4321#discussion_r463059042
https://github.com/sunpy/sunpy/pull/4321#discussion_r463059949
this all is for getting end times for a url. If file contains only `year` then end time is defaulted to end of that year.
If file contains only `y/m/d` then endtime is defaulted to end of the day,

It works for all files because none of the file have time in format `%j` or other format. In `SUVICLient` however the endtime was in the filename itself, so `post_search_hook` was overwritten.
All DR clients have daily files, except `SRS` has yearly file. But I think it is a bad assumption. Some scraper changes can be helpful to address it.